### PR TITLE
Add PostgreSQL test support

### PR DESCRIPTION
**Pull request by jlongever**
@RackHD/corecommitters @zyoung51 @uppalk1 @tannoa2 @keedya

### CODE REVIEW

File: vagrant/config/postgresql/config.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    "statsd": "127.0.0.1:8125",
+    "subnetmask": "255.255.252.0",
+    "syslogBindAddress": "172.31.128.1",
+    "syslogBindPort": 514,
+    "tftpBindAddress": "172.31.128.1",
+    "tftpBindPort": 69,
+    "tftpRoot": "./static/tftp",
+    "minLogLevel": 0,
+    "databaseOverrideTypes" : {
+        "catalogs": "postgresql",
```
Comment (yyscamper) [inline vagrant/config/postgresql/config.json:66]: why need to specify the data models one by one? Is their any user case that will store different kind data in different database?

Comment (yyscamper) [inline vagrant/config/postgresql/config.json:66]: get some answer from this PR(https://github.com/RackHD/RackHD/pull/267), but if this is a temporary state, seems it is not worth to add the example config file right now. We can update the config file after PostgreSQL become full ready.

Comment (jlongever) [inline vagrant/config/postgresql/config.json:66]: @amymullins @zyoung51 ^^ ??

Comment (zyoung51) [inline vagrant/config/postgresql/config.json:66]: @yyscamper This is a per-model override configuration and reflects the models that will work with PostgreSQL in the first iteration that adds support.  There is no use case that specifies that we should limit configurability of the product, but if you have a personal opinion that the product should be less configurable, I'm happy to remove this functionality and delay the PostgreSQL PR until sometime in the future when all the models are supported.

**Comment by heckj** [comment]
@yyscamper I don't expect this will be a temporary state - enabling more configuration options for deploying RackHD. @zyoung51 don't need to be passive aggressive about this, and felix's initial question is worthy of a real answer - what's the design thoughts behind breaking out each of the models into a configuration option.

**Comment by benbp** [comment]
I'm wondering the same thing, is there any case where we expect someone to actually want to use a different database per model? This seems to me like a case where adding more knobs just makes things more confusing and harder to understand.

**Comment by zyoung51** [comment]
@heckj The work is split into phases and only works in those models initially with the workflow related models to follow at a later date.  There is no data as to what a user would want to do or data on what we don't want them to do, so I erred on the side of enabling the user to do what they may want.  As I said, more than happy to take this out and wait for the whole thing to be ready rather than forcing in something that is too confusing.

**Comment by jlongever** [comment]
Looks like the functional spec needs to be revisited: https://github.com/RackHD/RackHD/wiki/(optional-configurable)-Persistence-Layers-for-RackHD

Closing for now and moving the story back to the backlog until we can determine the right approach.

**Comment by amymullins** [comment]
There is no plan to develop the entire feature and then submit the PRs.  That will be a very significant development effort and instead will be taking the approach of incremental development where small pieces of functionality will be developed and delivered each sprint.  

The support for the workflow store will be developed (have a story for that already in the backlog).  It  needs discussion as it requires pollers to be refactored and that is not a trivial effort.  The first story currently being worked on has not been completed and is considered a dependency for this particular story, so ok to close this PR and revisit when the dependency is removed.   

For reference, here is the proposal: https://github.com/RackHD/RackHD/wiki/(optional-configurable)-Persistence-Layers-for-RackHD

Dev activities can be referenced directly in the Gripen team backlog : https://www.pivotaltracker.com/n/projects/1492890

**Pull request closed without merging**